pure2.md

```diff
@@ -0,0 +1,535 @@
+
+When a devices receives online updates from the Director receive the latest version of the offline-update snapshot metadata as part of the normal update cycle.
+
+### Offline-update role in root metadata
+
+To perform offline updates, the root metadata file in the Director repository SHALL include a new "offline-updates" role. This role MAY use the same signing key as the Targets role, but SHOULD use its own. The offline-update role's key(s) MAY be kept online or offline, depending on the security requirements of the implementor.
+
+### Offline-update snapshot metadata
+
+The offline-update snapshot metadata lists version numbers and filenames of all offline-update targets metadata files.
+
+For each offline-update targets metadata file on the repository, the offline-update snapshot metadata SHALL contain the following information:
+
+1.  The filename and version number of the offline-update targets metadata file.
+
+### Offline-update targets metadata
+
+Each offline-update targets metadata file on a repository contains information about a set of images eligible to be installed on devices. Like other Director targets metadata files, offline-update targets metadata files SHALL NOT contain delegations.
+
+When a device is validating offline-update targets metadata, it will only execute an update if it can unambiguously determine which images to install. For this reason, the metadata SHOULD include a mechanism to permit disambiguation in multi-ECU systems. See the deployment considerations section for more detail on this recommendation.
+
+### Addendum to Director repository snapshot metadata requirements
+
+Offline-update targets metadata files SHOULD NOT be listed in the main snapshots metadata file of the Director repository, if implemented. Because the offline-update role signs its own snapshot metadata, listing those files in the main snapshots metadata file is redundant, and increases the size of a file that is downloaded very frequently.
+
+### Required files for offline updates
+
+The following elements SHALL be present on any offline-update storage location (in a "well-known" location configured in advance on the on the Primary):
+
+-   Image repository metadata:
+
+    -   All Root metadata files from the Image repository, from version 1 to the most-recent version at the time the offline update is generated
+
+    -   The latest version of every Targets metadata file, including delegations, at the time the offline update is generated
+
+    -   The latest version of the Snapshot metadata file at the time the offline update was generated
+
+-   Director repository metadata:
+
+    -   All Root metadata files from the Director repository, from version 1 to the most-recent version at the time the offline update is generated
+
+    -   The offline-update Snapshot metadata file
+
+    -   Exactly one offline-update targets metadata file
```

Comment:
What happens if there is  more than one?

Reply:
The client would take the first one listed in the snapshot metadata:

> 4. Traverse the list of offline-update targets metadata files in the offline-update Snapshot metadata until a matching filename is found in the well-known location.

I do think it might be useful to be able to include multiple options in one usb stick and have a user-controlled selection mechanism, but it would have complicated the spec to include user interaction/selection as part of the process. My thinking here was that an implementor could accommodate this use case just by making several different PURE-2 directories, and having the user select the one they want before the process starts, e.g.:

```
.well-known
├── option-1
│   ├── images
│   │   ├── firmware-acme-1.0.2.bin
│   │   └── firmware-bravo-3.1.1.bin
│   └── metadata
│       ├── director
│       │   ├── EMEA-standard.json
│       │   ├── offline-update-snapshot.json
│       │   ├── 1.root.json
│       │   ├── 2.root.json
│       │   ├── 3.root.json
│       │   ├── 4.root.json
│       │   └── 5.root.json
│       └── image-repo
│           ├── delegation-1.json
│           ├── delegation-2.json
│           ├── 1.root.json
│           ├── 2.root.json
│           ├── 3.root.json
│           ├── snapshot.json
│           └── targets.json
└── option-2
    ├── images
    │   ├── firmware-charlie-1.0.2.bin
    │   └── firmware-delta-3.1.1.bin
    └── metadata
        ├── director
        │   ├── APAC-standard.json
        │   ├── offline-update-snapshot.json
        │   ├── 1.root.json
        │   ├── 2.root.json
        │   ├── 3.root.json
        │   ├── 4.root.json
        │   └── 5.root.json
        └── image-repo
            ├── delegation-1.json
            ├── delegation-2.json
            ├── 1.root.json
            ├── 2.root.json
            ├── 3.root.json
            ├── snapshot.json
            └── targets.json
```